Tangem/App/Services/WalletConnect/WalletConnectV2Service/Handlers/SignHandling/WalletConnectV2SignTypedData.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  WalletConnectV2SignTypedData.swift
+//  Tangem
+//
+//  Created by Andrew Son on 30/01/23.
+//  Copyright © 2023 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import enum BlockchainSdk.Blockchain
+import struct WalletConnectSwiftV2.AnyCodable
+import enum WalletConnectSwiftV2.RPCResult
+
+struct WalletConnectV2SignTypedDataHandler {
+    private let message: String
+    private let typedData: EIP712TypedData
+    private let signer: WalletConnectSigner
+    private let walletModel: WalletModel
+
+    init(
+        requestParams: AnyCodable,
+        blockchain: Blockchain,
+        signer: WalletConnectSigner,
+        walletModelProvider: WalletConnectV2WalletModelProvider
+    ) throws {
+        let params = try requestParams.get([String].self)
+
+        guard params.count >= 2 else {
+            throw WalletConnectV2Error.notEnoughDataInRequest(requestParams.description)
+        }
+
+        message = params[1]
+
+        let targetAddress = params[0]
+        walletModel = try walletModelProvider.getModel(with: targetAddress, in: blockchain)
+        guard
+            let messageData = message.data(using: .utf8),
+            let typedData = try? JSONDecoder().decode(EIP712TypedData.self, from: messageData)
+        else {
+            throw WalletConnectV2Error.notEnoughDataInRequest(requestParams.description)
+        }
```

Comment:
```suggestion
        guard let messageData = message.data(using: .utf8) else { 
            throw WalletConnectV2Error.notEnoughDataInRequest(requestParams.description)
        }
        
        let typedData = try JSONDecoder().decode(EIP712TypedData.self, from: messageData)
```

а почему не так? что бы всегда ошибку из `requestParams.description` возвращать?

Reply:
да, тут важнее знать что за говно пришло в requestParams)